scalafix-rules/src/main/scala/scalafix/internal/rule/CompilerTypePrinter.scala

```diff
@@ -109,7 +109,26 @@ class CompilerTypePrinter(g: ScalafixGlobal, config: ExplicitResultTypesConfig)(
     }
     val preProcessed = preProcess(toLoop, inverseSemanticdbSymbol).widen
     val shortType = g.shortType(preProcessed, history)
-    val short = shortType.toString()
+    val shortTypeStr = shortType.toString()
+
+    // currently wrapping in backticks supported only for case classes
+    val isCaseClass = shortType.typeSymbol.isCaseClass
+    val typeSymbolName = shortType.typeSymbol.decodedName
```

Comment:
@bjaglin For example, I don't get why in case of traits `shortType.typeSymbol.decodedName` is producing `<none>`. Also, it was easy to break the current semantic when no backticks are needed  (function types, path-dependent types, etc).